src/NServiceBus.Core/Routing/RoutingFeature.cs

```diff
@@ -7,50 +7,50 @@
 
     class RoutingFeature : Feature
     {
-        public const string EnforceBestPracticesSettingsKey = "NServiceBus.Routing.EnforceBestPractices";
-
         public RoutingFeature()
         {
             EnableByDefault();
             Defaults(s =>
             {
-                s.SetDefault<UnicastRoutingTable>(new UnicastRoutingTable());
                 s.SetDefault<ConfiguredUnicastRoutes>(new ConfiguredUnicastRoutes());
-
-                s.SetDefault<EndpointInstances>(new EndpointInstances());
-                s.SetDefault<DistributionPolicy>(new DistributionPolicy());
-
                 s.SetDefault(EnforceBestPracticesSettingsKey, true);
-
-                s.SetDefault<Publishers>(new Publishers()); // required to initialize MessageEndpointMappings.
+                s.Set<RoutingFeature>(this);
```

Comment:
> If we follow this approach should a feature auto-register itself by default in the settings?

I like that idea

> context.Feature().Dependency

this seems to be the hardest one to realize, as it would require extension methods for every feature?

> context.Feature<Specific>().Dependency

I like that idea as well